src/byteVector.ts

```diff
@@ -624,6 +638,21 @@ export class ByteVector {
             : ByteVector.TD1;
     }
 
+    /**
+     * Inserts the contents of another ByteVector object into the current instance.
+     * @param index A number value specifying the index at which to insert the data.
+     * @param data A ByteVector object containing data to insert into the current instance.
+     */
+    public insert(index: number, data: ByteVector): void {
```

Comment:
This could indeed. However we'd have to perform the read-only check at the place of usage. In commit 748adf590351ea63543d561591d2d3528f32f784, I've chosen to keep insert, and instead cleaned it up to match other functions. However if you'd still prefer to to have it removed, please don't hesitate to tell me.

Reply:
> perform the read-only check at the place of usage

I'm a bit confused what you mean by that 🤔 `ByteVector.splice` does a check for read-only and if falsy/empty array is passed in, it'll do nothing. It looks like `insert` just omits the number of bytes to delete?

Reply:
You're right. I'm not sure how I missed that :) I've fixed it in 786086d47058b31cf92c6bcf2fecc47cfb22ea1f